plugin/cscope_maps.vim

```diff
@@ -41,12 +41,40 @@ if has("cscope")
     " if you want the reverse search order.
     set csto=0
 
-    " add any cscope database in current directory
-    if filereadable("cscope.out")
-        cs add cscope.out
-    " else add the database pointed to by environment variable
-    elseif $CSCOPE_DB != ""
+    " cscope seems to not index files in symlinked directories.
+    " However, one can build the cscope.files (list of files for cscope to index)
+    " via a command like:
+    " `find -L [your searched folder] -name [your searched pattern]
+    "   -exec realpath {} \; >> cscope.files`
+    "
+    " Note that it is prudent to use absolute file paths, hence the use of realpath.
+    "
+    " Now the trouble is getting vim to try to find a cscope.out file
+    " going recursively up through directories, but going back up through
+    " them via any symlinks you took to get to your CWD in the shell.
+    " The following codes solves this issue.
+
+    if $CSCOPE_DB != ""
         cs add $CSCOPE_DB
+    else
+        " Get all parts of out current path
+        let dirs = split($PWD, '/')
+        " Start building a lists of paths in which to look for cscope.out
+        let paths = ['/']
+        " /foo/bar/baz would result in the `paths` array containing:
+        " [/ /foo /foo/bar /foo/bar/baz]
+        for d in dirs
+            let paths = add(paths, paths[len(paths) - 1] . d . '/')
+        endfor
+
+        " List is backwords search order, so reverse it.
+        for d in reverse(paths)
+            let cscope_file = d . "/cscope.out"
+            if filereadable(cscope_file)
+                execute('cs add ' . cscope_file)
```

Comment:
What if there are several paths contain cscope.out file? Could there be a case when incorrect one is added?

Reply:
This finds the closest one. I suppose the user could want something else. We could always make this logic I'm adding enabled by some flag.

Reply:
Again, I'm probably missing the point... Can you please explain to me how can I reproduce the issue? Like, minimal possible example, so I can see why this is needed. It can be a bunch of bash commands to reproduce the issue, or some repo on GitHub. Sorry, it's just hard for me to see if the logic correct, when I don't understand the issue itself.